samcli/local/docker/lambda_build_container.py

```diff
@@ -163,6 +175,54 @@ def _get_container_dirs(source_dir, manifest_dir):
 
         return result
 
+    @staticmethod
+    def _convert_to_container_dirs(host_paths_to_convert, host_to_container_path_mapping):
+        """
+        Use this method to convert a list of host paths to a list of equivalent paths within the container
+        where the given host path is mounted. This is necessary when SAM CLI needs to pass path information to
+        the Lambda Builder running within the container.
+
+        If a host path is not mounted within the container, then this method simply passes the path to the result
+        without any changes.
+
+        Ex:
+            [ "/home/foo", "/home/bar", "/home/not/mounted"]  => ["/tmp/source", "/tmp/manifest", "/home/not/mounted"]
+
+        Parameters
+        ----------
+        host_paths_to_convert : list
+            List of paths in host that needs to be converted
+
+        host_to_container_path_mapping : dict
+            Mapping of paths in host to the equivalent paths within the container
+
+        Returns
+        -------
+        list
+            Equivalent paths within the container
+        """
+
+        if not host_paths_to_convert:
+            # Nothing to do
+            return host_paths_to_convert
+
+        # Make sure the key is absolute host path. Relative paths are tricky to work with because two different
+        # relative paths can point to the same directory ("../foo", "../../foo")
+        mapping = {str(pathlib.Path(p).resolve()): v for p, v in host_to_container_path_mapping.items()}
+
+        result = []
+        for original_path in host_paths_to_convert:
+            abspath = str(pathlib.Path(original_path).resolve())
+
+            if abspath in mapping:
```

Comment:
In other words, if the path that is supplied to be added as executable_search_path within the container is already present in the mapping of paths to be mounted, we add the path to be mounted within the container to list of executable search paths.

Also this is very opinionated to adding just source_dir or manifest_dir within the container to search paths, and not any other sub paths.

eg: instead of just /tmp/source_dir and /tmp/manifest_dir , we dont yet support /tmp/source_dir/myexecutable_versions/

Reply:
Correct, we don't support that. This is a good call out. We should be okay for the `gradlew` case now, but we definitely need to implement a generic path-translation logic that handles lot of these cases